packages/react-core/src/components/TextInput/examples/TextInputIconSprite.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { TextInput } from '@patternfly/react-core';
+
+export const TextInputIconSprite: React.FunctionComponent = () => {
+  const [calendar, setCalendar] = React.useState('');
+  const [clock, setClock] = React.useState('');
+
+  return (
+    <>
+      <TextInput validated={'success'} isIconSprite type="text" aria-label="success icon sprite text input example" />
+      <br />
+      <br />
+      <TextInput validated={'warning'} isIconSprite type="text" aria-label="warning icon sprite text input example" />
+      <br />
+      <br />
+      <TextInput validated={'error'} isIconSprite type="text" aria-label="error icon sprite text input example" />
```

Comment:
Initially it was because the states `success`, `warning`, and `error` were being unused - but I can just set the value of these fields as the states to work around that.